source/dagger/gameplay/platformer/platformer_controller.cpp

```diff
@@ -45,25 +53,75 @@ void PlatformerControllerSystem::Run()
                 }
             }
 
-            if (char_.isRolling)
+            if (jump == 1 && !char_.isRolling)
             {
-                if (char_.timeRolling < char_.rollingTime)
+                char_.isJumping = true;
+                if (run != 0)
                 {
-                    AnimatorPlay(animator_, "souls_like_knight_character:ROLL");
-                    sprite_.position.x += char_.rollingSpeed * sprite_.scale.x * Engine::DeltaTime();
-                    char_.timeRolling += Engine::DeltaTime();
+                    sprite_.scale.x = run;
                 }
-                else
+            }
+
+            if (char_.isRolling || char_.isJumping)
+            {
+
+                if (char_.isRolling)
+                {
+                    if (run != 0)
+                    {
+                        sprite_.scale.x = run;
+                    }
+                    if (char_.timeRolling < char_.rollingTime)
+                    {
+                        AnimatorPlay(animator_, "souls_like_knight_character:ROLL");
+                        sprite_.position.x += char_.rollingSpeed * sprite_.scale.x * Engine::DeltaTime();
+                        char_.timeRolling += Engine::DeltaTime();
+                    }
+                    else
+                    {
+                        char_.isRolling = false;
+                        char_.timeRolling = 0;
+                    }
+                }
+                else if (char_.isJumping)
                 {
-                    char_.isRolling = false;
-                    char_.timeRolling = 0;
+                    if (run != 0)
+                    {
+                        sprite_.scale.x = run;
+                    }
+                    Float32 speed = CalculateVerticalSpeed(char_.verticalInitialSpeed, char_.gravity, char_.timeJumping);
+                    if (!char_.reachedMax)
+                    {
+                        AnimatorPlay(animator_, "souls_like_knight_character:JUMP");
```

Comment:
Maybe we can add lines 95-99 in some function considering the fact that similar lines appear in the 105-109? And based on the direction of the movement we could play different animation?

Reply:
Yes. I actually intended to do this, but forgot :laughing: